src/omnicore/tx.cpp

```diff
@@ -411,30 +622,84 @@ bool CMPTransaction::interpret_MetaDExCancelEcosystem()
 /** Tx 50 */
 bool CMPTransaction::interpret_CreatePropertyFixed()
 {
-    if (pkt_size < 25) {
-        return false;
+    if (encodingClass == OMNI_CLASS_D) {
+        int i = 0;
+
+        std::vector<uint8_t> vecVersionBytes = GetNextVarIntBytes(i);
+        std::vector<uint8_t> vecTypeBytes = GetNextVarIntBytes(i);
+
+        memcpy(&ecosystem, &pkt[i], 1);
+        i++;
+
+        std::vector<uint8_t> vecPropTypeBytes = GetNextVarIntBytes(i);
+        std::vector<uint8_t> vecPrevPropIdBytes = GetNextVarIntBytes(i);
+
+        const char* p = i + (char*) &pkt;
+        std::vector<std::string> spstr;
+        for (int j = 0; j < 5; j++) {
+            spstr.push_back(std::string(p));
+            p += spstr.back().size() + 1;
+        }
+
+        if (isOverrun(p)) {
+            PrintToLog("%s(): rejected: malformed string value(s)\n", __func__);
+            return false;
+        }
+
+        int j = 0;
+        memcpy(category, spstr[j].c_str(), std::min(spstr[j].length(), sizeof(category)-1)); j++;
```

Comment:
Actually, we might ship a fix with class D encoding for "longer than allowed string sequences" here.

Right now it is possible to construct transactions, which have more characters than allowed in text fields. The extra characters are currently ignored, but it might be considered to invalidate such transactions.